src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/RecommendCaseInsensitiveStringComparison.Fixer.cs

```diff
@@ -0,0 +1,201 @@
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.NetCore.Analyzers.Performance
+{
+    /// <summary>
+    /// CA1862: Prefer the StringComparison method overloads to perform case-insensitive string comparisons.
+    /// </summary>
+    public class RecommendCaseInsensitiveStringComparisonFixer : CodeFixProvider
+    {
+        public override ImmutableArray<string> FixableDiagnosticIds { get; } = ImmutableArray.Create(RecommendCaseInsensitiveStringComparisonAnalyzer.RuleId);
+        public sealed override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            CancellationToken ct = context.CancellationToken;
+            if (ct.IsCancellationRequested)
+            {
+                return;
+            }
+
+            Document doc = context.Document;
+
+            SyntaxNode root = await doc.GetSyntaxRootAsync(ct).ConfigureAwait(false);
+
+            if (root.FindNode(context.Span, getInnermostNodeForTie: true) is not SyntaxNode node)
+            {
+                return;
+            }
+
+            SemanticModel model = await doc.GetSemanticModelAsync(ct).ConfigureAwait(false);
+
+            if (model.GetOperation(node, ct) is not IInvocationOperation invocation)
+            {
+                return;
+            }
+
+            if (model.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemStringComparison)
+                is not INamedTypeSymbol stringComparisonType)
+            {
+                return;
+            }
+
+            if (model.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemStringComparer)
+                is not INamedTypeSymbol stringComparerType)
+            {
+                return;
+            }
+
+            // Only diagnose overloads that take one string argument:
+            // Contains(string), IndexOf(string), StartsWith(string), CompareTo(string)
+            if (invocation.Arguments.Count() != 1)
+            {
+                return;
+            }
+
+            // There should be a child invocation on the left side (instance) of the large invocation
+            // If the large invocation is "a.ToLower().Contains(b)"
+            // Its instance is "a.ToLower()", the child invocation
+            if (invocation.Instance is not IInvocationOperation instanceOperation)
+            {
+                return;
+            }
+
+            string caseChangingMethodName;
```

Comment:
caseChangingPropertyName?

Reply:
Maybe I should use a different word. I was using "method" here as synonym of "technique" or "approach", not referring to "source code function".

The case changing string stored in this variable is used to retrieve either enum values from `StringComparison` or the properties from `StringComparer`.

I'll change this to `caseChangingApproachName`.